app/view_utils/path_helpers.rb

```diff
@@ -56,7 +56,7 @@ def landing_page_path(community_id:, logged_in:, locale_param:, default_locale:)
   end
 
   def paths
-    RequestStore.store[:url_helpers] ||= Rails.application.routes.url_helpers
+    @_url_helpers ||= Rails.application.routes.url_helpers
```

Comment:
Yes, it doesn't have to, there's nothing request specific in it. And the autoloading in dev mode still works as intended because of how Rails autoloads all code when it detects a change e.g. in routes.rb.